src/main/java/de/tum/in/www1/artemis/web/rest/dto/ProgrammingExerciseGitDiffEntryDTO.java

```diff
@@ -0,0 +1,24 @@
+package de.tum.in.www1.artemis.web.rest.dto;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import de.tum.in.www1.artemis.domain.hestia.ProgrammingExerciseGitDiffEntry;
+
+/**
+ * DTO for a git diff report entry.
+ *
+ * @param previousFilePath  the previous file path of the entry
+ * @param filePath          the new file path of the entry
+ * @param previousStartLine the previous start line of the entry
+ * @param startLine         the new start line of the entry
+ * @param previousLineCount the previous line count of the entry
+ * @param lineCount         the new line count of the entry
+ */
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+public record ProgrammingExerciseGitDiffEntryDTO(String previousFilePath, String filePath, Integer previousStartLine, Integer startLine, Integer previousLineCount,
+        Integer lineCount) {
+
+    public ProgrammingExerciseGitDiffEntryDTO(ProgrammingExerciseGitDiffEntry entry) {
```

Comment:
Discuss this with @MaximilianAnzinger 😁
IMO it's also fine to be not consistent here because for the CommitInfoDTO using the constructor approach would have clear disadvantages.
@MaximilianAnzinger What's actually your rationale for disliking these static methods?
The Java SE classes use this approach quite frequently.